pkg/gps/GpsProducer.php

```diff
@@ -27,16 +27,20 @@ public function __construct(GpsContext $context)
     }
 
     /**
-     * @param GpsTopic   $destination
-     * @param GpsMessage $message
+     * @param GpsTopic|GpsQueue $destination
+     * @param GpsMessage        $message
      */
     public function send(Destination $destination, Message $message): void
     {
-        InvalidDestinationException::assertDestinationInstanceOf($destination, GpsTopic::class);
+        $destination instanceof \Interop\Queue\Topic
+            ? InvalidDestinationException::assertDestinationInstanceOf($destination, GpsTopic::class)
+            : InvalidDestinationException::assertDestinationInstanceOf($destination, GpsQueue::class)
+        ;
         InvalidMessageException::assertMessageInstanceOf($message, GpsMessage::class);
 
+        $gpsTopicName = $destination instanceof GpsTopic ? $destination->getTopicName() : $destination->getQueueName();
         /** @var Topic $topic */
-        $topic = $this->context->getClient()->topic($destination->getTopicName());
+        $topic = $this->context->getClient()->topic($gpsTopicName);
```

Comment:
Sorry, It does not make much sense to me. You use queue but publish to the topic. Could you elaborate on this?

Reply:
I am not much into gps. But I recall that it is not possible to publish to gps queue directly, hence it has not been implemented.

Reply:
As far as I understand gps doesn't have a concept of queues, just topics and subscriptions. Hence I'd treat topic as a queue here. The problem is when clients router processor sends a message it sends it to the queue.

Reply:
I'd like to hear more opinions on this. Especially whether the topic and queue are the same in Google PubSub world or not. 

I am not an expert.

Reply:
Some links to refs or docs that prove it is same stuff would be ok too

Reply:
There’s just no such thing as a queue in the gps. You’d always publish messages to a topic. Here’s some more explanation: https://cloud.google.com/pubsub/docs/overview#concepts

Reply:
I've been digging into gps transport code to get better understanding of the problem. 
I can see that currently we treat the pubsub topic as enqueue topic and pubsub subscription as queue. And that would be fine if not for for the fact that there are cases where enqueue tries to send a message to the queue. 

As is it not possible to publish anything to the pubsub subscription I believe we could treat the enqueue queue in two ways:
- as a topic for publishing messages
- as a queue for procsessing messages
(as per this PR)

Reply:
bump

Reply:
@makasim , If I may join the discussion - the change @biforb4 did is correct. I've just spent the whole day digging through enqueue and his suggestion is correct. 
We are switching from AWS to GCloud and without the change, enqueue is basically useless for GPS.
I really hope you will reconsider this PR.